src/main/java/com/github/alexcojocaru/mojo/elasticsearch/v2/step/ResolveElasticsearchStep.java

```diff
@@ -34,6 +38,7 @@ public void execute(InstanceConfiguration config)
             unpackDirectory = unpackToElasticsearchDirectory(artifact, config);
 
             setupElasticsearchConf(config);
+            keepExistingDataCheck(config);
```

Comment:
Instead of wiping out in this step, could you create a separate step and add it to the InstanceSetupSequence? That way, you could easily unit test it - which is another ask for you.